content/tutorials/react-admin-panel-with-ably-and-refine.textile

```diff
@@ -0,0 +1,289 @@
+
+```[nodejs]
+...
+//highlight-start
+import { ablyClient } from "utility";
+import { liveProvider } from "@refinedev/ably";
+//highlight-end
+
+function App() {
+  return (
+          ...
+            <Refine
+              ...
+              //highlight-next-line
+              liveProvider={liveProvider(ablyClient)}
+              options={{
+              ...
+                //highlight-next-line
+                liveMode: "auto",
+              }}
+            >
+            ...
+  );
+}
+
+export default App;
+```
+
+h3. liveMode
+
+Quick info for live mode options.
+
+@auto@: Queries of related resource are invalidated in Realtime as new events from subscription arrive.
+
+@manual@: Queries of related resource are not invalidated in Realtime, instead "onLiveEvent":https://refine.dev/docs/api-reference/core/providers/live-provider/#onliveevent is run with the event as new events from subscription arrive.
+
+@off@: Disables live mode.
+
+"Refer to live provider docs for more information":https://refine.dev/docs/api-reference/core/providers/live-provider/
+
+h3. We are ready for testion realtime feature on the app
+
+That is everything you need to do to integrate Ably with your refine application.
+
+You can now perform CRUD operations on the @blog_posts@ in the fake API we used. When you delete, update, or create a new product, Ably will broadcast the changes in realtime.
+
+As demonstrated below, any modifications made to an existing record are immediately reflected, all thanks to the Ably live provider.
+
+<a href="/images/tutorials/ably-refine/img4.gif" target="_blank">
+  <img src="/images/tutorials/ably-refine/img4.gif" style="width: 100%"
+  title="Click to enlarge" alt="The realtime dashboard working">
+</a>
+
+This tutorial illustrates how you can build a basic CRUD app with realtime feature with refine and Ably. There is a lot you can do with refine and Ably that I have not highlighted in this article. Check out the "refine":https://refine.dev/docs/ or "Ably":https://www.ably.com/docs documentation.
+
+h2. Conclusion
+
+As explained above, you need considerable resources to build a complete data-intensive realtime platform from the ground up. However, you can also quickly bootstrap a complete realtime application with React frameworks such as refine and cloud-based platforms like Ably.
+
+As highlighted above, refine is a feature-rich React-based framework. It is an excellent choice, especially when building data-intensive front-end applications. The refine ecosystem has features for integrating some cloud-based solutions, such as Ably, into your front-end application.
+
+You can quickly build realtime solutions with refine and Ably’s distributed network of reliable, low-latency data centers. As highlighted above, getting started with refine and Ably is relatively straightforward.
```

Comment:
This states "as explained above" or "as highlighted above" three times. All three paragraphs are saying the same thing too. I wonder if this page isn't required and the source code can just sit under the previous heading. WDYT?